### PR TITLE
Ndcbw4d3 add service name to billing report

### PR DESCRIPTION
**Pull request by chrisclayson**
## What

We often need to know which service to associate a billable event with. We do this by running the 'verifications by RP' report, which uses some data from the billing report but adds in a column containing the service name for each report.

We could just include that column in the billing report.

## Why

We could stop running the verifications by RP report. 

Secondary reports that use that report could be simplified.

## How

Change `write_billing_event_to_database()` to also write `transaction_entity_id` from original event details.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,7 +14,14 @@ Running pre-commit will start docker-compose with Postgres and our code then run
 
 The database scripts live in [verify-event-system-database-scripts](https://github.com/alphagov/verify-event-system-database-scripts).
 
-The tests currently rely on scripts in the migrations directory of the above repo.
+The `./build/run-tests.sh` script expects to find the above repo, checked out, in the `../verify-event-system-database-scripts`
+directory. If it does not find it there, it will automatically clone it to that location.
```
Comment (galund) [inline README.md]: This feels a bit weird. I still kinda wonder whether the schema and the things that effectively depend on it should be in a mono-repo.

File: build/run-tests.sh [hunk 2]
```diff
@@ -1,7 +1,25 @@
...
+    echo "Cloning database migration scripts repo..."
+    pushd ..
+    git clone git@github.com:alphagov/verify-event-system-database-scripts.git
+    popd
+fi
+
+echo "Building test imgages..."
+docker-compose build
+
+echo "Starting up database..."
```
Comment (galund) [inline build/run-tests.sh]: This might wanna be a separate PR, but let's talk tomorrow :-)

File: src/database.py [hunk 3]
```diff
@@ -74,14 +85,20 @@ def write_billing_event_to_database(event, db_connection):
...
+                event.details['transaction_entity_id']
             ])
     except KeyError as keyError:
         getLogger('event-recorder').warning('Failed to store a billing event [Event ID {0}] due to key error'.format(event.event_id))
         raise keyError
     except IntegrityError as integrityError:
-        getLogger('event-recorder').warning('Failed to store a billing event [Event ID {0}] due to integrity error'.format(event.event_id))
-        raise integrityError
+        if integrityError.pgcode == UNIQUE_VIOLATION:
+            # The event has already been recorded - don't throw an exception (no need to retry this message), just
```
Comment (galund) [inline src/database.py]: Does this happen? Not clear why it's part of this PR.

**Comment by chrisclayson** [comment]
[Add service name to billing report](https://trello.com/c/NDCBW4d3/21-add-service-name-to-billing-report)

File: README.md (hunk 1 above)
Comment (chrisclayson) [reply README.md]: We moved to its own repo last quarter, the schema used to be in this repo.

I agree it's not ideal, but there is precedent for this with the hub tests relying on different repos being in specific locations.

We have two repos that are reliant on the schema migrations (the reporter and the recorder), and we'll need to make similar changes to the reporter repo, as those tests will have similar issues now too.

File: build/run-tests.sh (hunk 2 above)
Comment (chrisclayson) [reply build/run-tests.sh]: This is actually part of https://github.com/alphagov/verify-event-recorder-service/pull/57
